### PR TITLE
[ode] Fix compile ode in release only

### LINKED ISSUE #30433: [ode] build failure x64-windows-release
**Host Environment**

- OS: windows 11
- Compiler: vs 2022

**To Reproduce**

`./vcpkg install ode --triplet=x64-windows-release`

**Failure logs**

```
Installing 1/37 ode:x64-windows-release...
Building ode[core]:x64-windows-release...
warning: -- Using community triplet x64-windows-release. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: C:\src\vcpkg\triplets\community\x64-windows-release.cmake
-- Installing port from location: C:\src\vcpkg\buildtrees\versioning_\versions\ode\254c6b22b73ff9ce79f38f282f61cd85ca590128
-- Using cached odedevs-ode-0.16.3.tar.gz.
-- Cleaning sources at C:/src/vcpkg/buildtrees/ode/src/0.16.3-3c7e622114.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/src/vcpkg/downloads/odedevs-ode-0.16.3.tar.gz
-- Using source at C:/src/vcpkg/buildtrees/ode/src/0.16.3-3c7e622114.clean
-- Found external ninja('1.11.0').
-- Configuring x64-windows-release
-- Building x64-windows-release-rel
CMake Error at scripts/cmake/vcpkg_replace_string.cmake:2 (file):
  file failed to open for reading (No such file or directory):

    C:/src/vcpkg/packages/ode_x64-windows-release/debug/bin/ode-config
Call Stack (most recent call first):
  buildtrees/versioning_/versions/ode/254c6b22b73ff9ce79f38f282f61cd85ca590128/portfile.cmake:24 (vcpkg_replace_string)
  scripts/ports.cmake:147 (include)
```
- Cut and paste the appropriate build messages from the console output.
- Attach ***every failure log file*** mentioned in the console output.

**Additional context**

Add any other context about the problem here, such as what you have already tried to resolve the issue.
I guess it expect for debug folder. Not sure why.

### PR DESCRIPTION
**Pull request by talregev**
Fix compile ode in release only
fix #30433 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

You are free modify this port as you see fit without asking me.

### CODE REVIEW

File: ports/ode/portfile.cmake [hunk 1]
```diff
@@ -20,13 +20,22 @@ vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/ode-${VERSION})
 if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
     file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
 else()
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/bin/ode-config" "${CURRENT_PACKAGES_DIR}" "`dirname $0`/..")
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/bin/ode-config" "${CURRENT_PACKAGES_DIR}" "`dirname $0`/../..")
```
Comment (dg0yt) [inline ports/ode/portfile.cmake:24]: This is the only line which needs to be guarded. Use `if(NOT VCPKG_BUILD_TYPE)`. The only possible value (if not empty) is `release` at this point.

File: ports/ode/portfile.cmake [hunk 2]
```diff
@@ -20,13 +20,22 @@ vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/ode-${VERSION})
...
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib/cmake")
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/cmake")
+
+if(VCPKG_BUILD_TYPE STREQUAL "debug")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/lib/cmake")
+endif()
+if (VCPKG_BUILD_TYPE STREQUAL "release")
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/lib/cmake")
```
Comment (dg0yt) [inline ports/ode/portfile.cmake:38]: Is there CMake config? Why does it need to be removed?

**Review by dg0yt — COMMENTED** [review]
The new conditions break normal builds (`VCPKG_BUILD_TYPE` unset).

File: ports/ode/portfile.cmake (hunk 2 above)
Comment (talregev) [reply ports/ode/portfile.cmake]: I don't know.

File: ports/ode/portfile.cmake (hunk 1 above)
Comment (talregev) [reply ports/ode/portfile.cmake:24]: Thank you for the tips! I change it, hope it work now.

Comment (talregev) [reply ports/ode/portfile.cmake:24]: I need to do `if(NOT VCPKG_BUILD_TYPE)` on both line? Can you take a look on my solution. Is it similar? It not working.

Comment (talregev) [reply ports/ode/portfile.cmake:24]: I figure it out. Thank you!.

**Review by LilyWangLL — APPROVED** [review]

**Comment by talregev** [comment]
Can this PR will merge today? This is a simple fix.

**Pull request merged**